### PR TITLE
test: Add a min-llvm-version directive

### LINKED ISSUE #36138: Travis is failing
I believe this is due to travis using system llvm which misses the fix backported in #36117.

```
---- [run-pass] run-pass/issue-36023.rs stdout ----
	
error: test run failed!
status: signal: 11
command: x86_64-unknown-linux-gnu/test/run-pass/issue-36023.stage2-x86_64-unknown-linux-gnu 
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------

------------------------------------------

thread '[run-pass] run-pass/issue-36023.rs' panicked at 'explicit panic', /build/src/tools/compiletest/src/runtest.rs:2356
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

cc @eddyb

### PR DESCRIPTION
**Pull request by alexcrichton**
We've got tests which require a particular version of LLVM to run as they're
testing bug fixes. Our build system, however, supports multiple LLVM versions,
so we can't run these tests on all LLVM versions.

This adds a new `min-llvm-version` directive for tests so they can opt out of
being run on older versions of LLVM. This then namely applies that logic to the
`issue-36023.rs` test case and...

Closes #36138

### CODE REVIEW

**Comment by alexcrichton** [comment]
(note that this should hold off on r+ until we confirm travis is green)

**Comment by frewsxcv** [comment]
https://travis-ci.org/rust-lang/rust/builds/156884048#L7287

```
test [run-pass] run-pass/issue-36023.rs ... ignored
```

File: src/tools/compiletest/src/header.rs [hunk 1]
```diff
@@ -115,6 +117,24 @@ impl EarlyProps {
...
+        fn ignore_llvm(config: &Config, line: &str) -> bool {
+            if let Some(ref actual_version) = config.llvm_version {
+                if line.contains("min-llvm-version") {
+                    let min_version = line.trim()
+                        .split(' ')
+                        .last()
+                        .expect("Malformed llvm version directive");
+                    // Ignore if actual version is smaller the minimum required
+                    // version
+                    &actual_version[..] < min_version
```
Comment (TimNN) [inline src/tools/compiletest/src/header.rs]: Won't this fail for something like llvm version 3.10?

Comment (brson) [inline src/tools/compiletest/src/header.rs]: Looks like it.

Comment (brson) [inline src/tools/compiletest/src/header.rs]: But LLVM 3.9 is going to be followed by 4.0 so maybe it doesn't matter.

File: src/bootstrap/check.rs [hunk 2]
```diff
@@ -148,6 +148,8 @@ pub fn compiletest(build: &Build,
     if let Some(ref dir) = build.lldb_python_dir {
         cmd.arg("--lldb-python-dir").arg(dir);
     }
+    let llvm_version = output(Command::new(&llvm_config).arg("--version"));
```
Comment (frewsxcv) [inline src/bootstrap/check.rs]: Should this be `build.llvm_config`?

Comment (frewsxcv) [inline src/bootstrap/check.rs]: Regardless of what it *should* be, `llvm_config` is undefined at this point:

```
error[E0425]: unresolved name `llvm_config`
   --> src/bootstrap/check.rs:151:45
    |
151 |     let llvm_version = output(Command::new(&llvm_config).arg("--version"));
    |                                             ^^^^^^^^^^^
```

**Comment by sanxiyn** [comment]
LLVM 3.9.0 release did not include r279980, so the test actually fails with LLVM 3.9.0, although it succeeds with Rust's LLVM fork.

**Comment by eddyb** [comment]
@sanxiyn I'm sorry I didn't get them to backport the patches :disappointed:.

**Pull request merged**